### PR TITLE
Fix yang model for ICMP and ICMPV6

### PR DESCRIPTION
**Pull request by bingwang-ms**
#### Why I did it
Fix[ #18137](https://github.com/sonic-net/sonic-buildimage/issues/18137)

This PR is to update the definition of Yang model for ICMP and ICMPv6.

|             | Before change | After change |
|-------------|---------------|--------------|
| ICMP type   |    [1, 44]    |    [0, 44]   |
| ICMP code   |    [1, 16]    |    [0, 16]   |
| ICMPV6 type |    [1, 44]    |   [0, 255]   |
| ICMPV6 code |    [1, 44]    |   [0, 255]   |

References
1. https://datatracker.ietf.org/doc/html/rfc792
2. https://datatracker.ietf.org/doc/html/rfc4443
##### Work item tracking
- Microsoft ADO **27118551**:

#### How to verify it
Verified by UT.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [x] 202305
- [x] 202311 

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ] 

#### Description for the changelog

 Update the definition of Yang model for ICMP and ICMPv6.

### CODE REVIEW

**Review by Blueve — APPROVED** [review]

**Review by lizhijianrd — APPROVED** [review]

File: src/sonic-yang-models/yang-templates/sonic-acl.yang.j2 [hunk 1]
```diff
@@ -202,13 +202,13 @@ module sonic-acl {
 						when "not(IP_TYPE) or boolean(IP_TYPE[.='ANY' or .='IP' or .='IPV4' or .='IPv4ANY' or .='ARP'])";
 						leaf ICMP_TYPE {
 							type uint8 {
-								range 1..44;
+								range 0..44;
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-templates/sonic-acl.yang.j2:205]: Maximum value is 44, why do we have this limit?

Comment (bingwang-ms) [reply src/sonic-yang-models/yang-templates/sonic-acl.yang.j2:205]: Yes, maximun value is 44. We have this limit.

Comment (ganglyu) [reply src/sonic-yang-models/yang-templates/sonic-acl.yang.j2:205]: https://www.iana.org/assignments/icmp-parameters/icmp-parameters.xhtml
https://www.iana.org/assignments/icmpv6-parameters/icmpv6-parameters.xml
Please check above links, do we need to modify the type range and code range?

Comment (bingwang-ms) [reply src/sonic-yang-models/yang-templates/sonic-acl.yang.j2]: Updated.

**Review by ganglyu — APPROVED** [review]

**Review by wen587 — APPROVED** [review]

**Pull request merged**